sql/moz-fx-data-shared-prod/telemetry/active_users/view.sql

```diff
@@ -0,0 +1,189 @@
+CREATE OR REPLACE VIEW
+  `moz-fx-data-shared-prod.telemetry.active_users`
+AS
+-- Distribution_id is only collected for Desktop and Fenix as of Q1/2024.
+-- See e.g. Firefox iOs https://dictionary.telemetry.mozilla.org/apps/firefox_ios?page=1&search=distribution_id
+-- TODO: Integrate distribution_id in fenix.firefox_android_clients & backfill to use it here.
+WITH fenix_distribution_id AS (
+  SELECT
+    client_info.client_id AS client_id,
+    sample_id AS sample_id,
+    DATE(submission_timestamp) AS submission_date,
+    ARRAY_AGG(
+      metrics.string.metrics_distribution_id IGNORE NULLS
+      ORDER BY
+        submission_timestamp ASC
+    )[SAFE_OFFSET(0)] AS distribution_id,
+  FROM
+    `moz-fx-data-shared-prod.fenix.baseline`
+  GROUP BY
+    client_id,
+    sample_id,
+    submission_date
+)
+-- Firefox Desktop
+SELECT
+  submission_date,
+  client_id,
+  CASE
+    WHEN isp_name = 'BrowserStack'
+      THEN CONCAT(app_name, ' ', isp_name)
+    WHEN distribution_id = 'MozillaOnline'
+      THEN CONCAT(app_name, ' ', distribution_id)
+    ELSE app_name
+  END AS app_name,
+  days_seen_bits,
+  days_active_bits,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) = 0 AS BOOLEAN) AS is_dau,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) <= 7 AS BOOLEAN) AS is_wau,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) AS BOOLEAN) AS is_mau,
+  CAST(mozfun.bits28.days_since_seen(days_seen_bits) = 0 AS BOOLEAN) AS is_daily_user,
+  CAST(mozfun.bits28.days_since_seen(days_active_bits) <= 7 AS BOOLEAN) AS is_weekly_user,
+  CAST(mozfun.bits28.days_since_seen(days_seen_bits) AS BOOLEAN) AS is_monthly_user,
+  CAST(1 AS BOOLEAN) AS is_desktop,
```

Comment:
Why not `TRUE AS is_desktop`? Not a big deal, just curious.

Reply:
Actually, I think we probably want something like this logic to filter out MozillaOnline and BrowserStack:

```suggestion
    IF(LOWER(isp) <> "browserstack" AND LOWER(distribution_id) <> "mozillaonline", TRUE, FALSE) AS is_desktop,
```

Reply:
Do we want this pre-filtered for all apps?
Desktop and Fenix filter out BrowserStack & MzoillaOnline
and other apps filter out BrowserStack?

Reply:
E.g. Reporting the MozillaOnline DAU would require filtering by app_name. Sounds ok?

Reply:
If by pre-filter you mean filtering via the `WHERE` clause in each subquery, we don't want to do that; we need to use `app_name` to label BrowserStack for all apps and MozillaOnline where possible. The only `app_names` that should be marked as `is_mobile` are "Fenix", "Firefox iOS", "Focus Android", and "Focus iOS". In desktop, only "Firefox Desktop" should be marked as `is_desktop`.

In the final view, we should be able to count MozillaOnline DAU using:

```sql
SELECT submission_date,
       app_name,
       SUM(is_dau) AS dau
  FROM active_users
 WHERE LOWER(app_name) LIKE "%mozillaonline"
 GROUP BY 1, 2
 ORDER BY 1
```